src/api/app/models/kiwi/image.rb

```diff
@@ -144,6 +144,24 @@ def default_package_group
     package_groups.find_or_create_by(kiwi_type: :image, pattern_type: 'onlyRequired')
   end
 
+  def self.find_binaries_by_name(query, project, use_project_repositories, repositories)
```

Comment:
Instead of using `true/false` for `use_project_repositories` I would rather see something like:
```
def self.find_binaries_by_name(query, project, repositories, opts = {})
```
so that you provide is as a hash like this:
```
self.find_binaries_by_name(query, project, repositories, use_project_repositories: true)
```

This makes it more expressive.

Reply:
Ok, will do... but this is not optional...